core/src/wsv.rs

```diff
@@ -33,53 +28,54 @@ pub type NewBlockNotificationSender = tokio::sync::watch::Sender<()>;
 /// Receiver type of the new block notification channel
 pub type NewBlockNotificationReceiver = tokio::sync::watch::Receiver<()>;
 
-/// World trait for mocking
-pub trait WorldTrait:
-    Deref<Target = World> + DerefMut + Send + Sync + 'static + Debug + Default + Sized + Clone
-{
-    /// Creates a [`World`] with these [`Domain`]s and trusted [`PeerId`]s.
-    fn with(
-        domains: impl IntoIterator<Item = Domain>,
-        trusted_peers_ids: impl IntoIterator<Item = PeerId>,
-    ) -> Self;
-}
-
 /// The global entity consisting of `domains`, `triggers` and etc.
 /// For example registration of domain, will have this as an ISI target.
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, Getters)]
+#[cfg_attr(feature = "mock", allow(dead_code))]
+#[getset(get = "pub")]
```

Comment:
I'm very much against using getters/setters in rust unless it's not possible to get around. This pattern is just not well suited for Rust because you loose the ability to transfer ownership or destructure or take disjoint references, etc. It was a different case in the `data_model` because there getters/setters are required to translate into FFI calls in the background

Reply:
I totally agree. Furthermore, since scope can be tightly controlled, the `getset(get = "pub")` directive can really be replaced with `pub(crate)` or `pub(super)` for the fields.

Reply:
Never thought that it's a bad idea in Rust...

Reply:
I understand that we loose to much things like moving, I'll remove it, this getters were used for mocking.
 Although without getters it feels for me like we have no encapsulation

Reply:
Encapsulation =/= getters/setters. 

In rust, the main point is gone because visibility modifiers, interior mutability and borrow checking won't allow you to break any invariants. The only use case that is not covered is dynamic dispatch in dynamic linking, which is only used for data_model entities.